solidity/test/GovernanceTest.js

```diff
@@ -180,50 +196,88 @@ describe("TBTCSystem governance", async function() {
       await restoreSnapshot()
     })
 
-    it("does not revert if beginKeepFactorySingleShotUpdate has already been called", async () => {
-      await tbtcSystem.beginKeepFactorySingleShotUpdate(
+    it("does not revert if beginKeepFactoriesUpdate has already been called", async () => {
+      await tbtcSystem.beginKeepFactoriesUpdate(
         "0x0000000000000000000000000000000000000001",
         "0x0000000000000000000000000000000000000002",
+        "0x0000000000000000000000000000000000000003",
       )
 
+      const finalizationTime = await tbtcSystem.getRemainingKeepFactoriesUpdateTime()
+      await increaseTime(finalizationTime.subn(1))
+
       // Should not revert.
-      await tbtcSystem.beginKeepFactorySingleShotUpdate(
+      await tbtcSystem.beginKeepFactoriesUpdate(
+        "0x0000000000000000000000000000000000000005",
+        "0x0000000000000000000000000000000000000006",
+        "0x0000000000000000000000000000000000000007",
+      )
+
+      expect(await tbtcSystem.getRemainingKeepFactoriesUpdateTime()).to.eq.BN(
+        await tbtcSystem.getGovernanceTimeDelay(),
+      )
+    })
+
+    it("does not revert if finalizeKeepFactoriesUpdate has already been called", async () => {
```

Comment:
`governanceTest` covers just a single update. Here we wanted to check if it's possible to perform multiple updates.

In `does not revert if beginKeepFactoriesUpdate has already been called` we wanted to confirm that a value can be overwritten for the ongoing update process before it's finalized.

In `does not revert if finalizeKeepFactoriesUpdate has already been called` we want to check that it's possible to update a value for a second time.

Reply:
`does not revert if beginKeepFactoriesUpdate has already been called` is covered by the `overrides previous update and resets timer` test in `governanceTest`.

It does look like we're missing the second one… We probably should test that in `governanceTest` in that case, since it's a shared property. But that's for another day.

Reply:
I covered them in https://github.com/keep-network/tbtc/pull/725